utils/infura.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// SendRequest will send a request to infura and return the unmarshalled data
+// back to the caller. It will retry until a valid response is returned, or
+// until the context times out.
+func SendRequest(ctx context.Context, url string, request string) ([]byte, error) {
+
+	// Retry until a valid response is returned or until context times out
+	for {
```

Comment:
@divyakoshy this should use the standard back-off retry with a maximum delay of 30s.

Reply:
Corrected and pushed.